Movecraft/src/main/java/net/countercraft/movecraft/Movecraft.java

```diff
@@ -244,6 +237,52 @@ public void onEnable() {
         logger.info("[V " + getDescription().getVersion() + "] has been enabled.");
     }
 
+    private void initializeCommands() {
+        MovecraftCommandManager movecraftCommandManager = new MovecraftCommandManager(this);
+        var commandCompletions = movecraftCommandManager.getCommandCompletions();
+
+        commandCompletions.registerCompletion("crafttypes", c ->  {
+            Set<CraftType> craftTypes = CraftManager.getInstance().getCraftTypes();
+            List<String> craftNames = craftTypes.stream().map(type -> type.getStringProperty(CraftType.NAME)).toList();
+            return craftNames;
+        });
+
+        commandCompletions.registerCompletion("directions", c -> {
+            var allDirections = CruiseDirection.valuesString();
+            var allButNone = allDirections.stream().filter(p -> !p.equals("none")).toList();
+            return allButNone;
+        });
+
+        var commandContexts = movecraftCommandManager.getCommandContexts();
+
+        commandContexts.registerContext(CruiseDirection.class, (c) -> {
+            String data = c.popFirstArg();
+            return CruiseDirection.fromString(data);
+        });
+
+        commandContexts.registerContext(CraftType.class, (c) -> {
+            String data = c.popFirstArg();
+            CraftType type = CraftManager.getInstance().getCraftTypeFromString(data);
+
+            if (type == null) {
+                throw new InvalidCommandArgument("You must supply a craft type!");
+            }
+
+            return type;
+        });
```

Comment:
Can we do all of this in the init of each command?

Reply:
I don't think so, if they are a pain to read there i can relocate it to its own functon, once a context is registered you can use it in any command

Reply:
Can it be put into the command manager instead?